source/plugin/event/causes.rst

```diff
@@ -0,0 +1,121 @@
+============
+Event Causes
+============
+
+Events are great for the use case of wanting to execute something when certain actions occur, but
+they have the drawback of providing next to no context as to what has **caused** that event to
+occur. A ``Cause`` allows providing and receiving additional contextual information about the event
+which can be do to modify the behavior of your event handler depending on this context.
```

Comment:
Something is wrong with this sentence, "which can be do to modify". We can clean up some wording later, but I'm really not sure what you meant to say here.